### PR TITLE
Add CustomEvent constructor polyfill

### LINKED ISSUE #8312: ReferenceError: CustomEvent is not defined
`CustomEvent` only has [partial support on IE](http://caniuse.com/#feat=customevent) and has generated several hundred errors on prod so far. We should:

- Add a helper method that uses `document.createElement('CustomEvent')` when `CustomEvent` constructor is not available.
- Add a presubmit check for direct usage of `CustomEvent()`.

### PR DESCRIPTION
**Pull request by choumx**
Fixes #8312.

IE doesn't support `CustomEvent` constructor. Pretty simple logic but let me know if you think a unit test is worthwhile.

/to @jridgewell @kmh287

### CODE REVIEW

File: src/utils/event.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Returns a CustomEvent with a given type and detail; supports fallback for IE.
+ * @param {!Window} win
+ * @param {string} type
+ * @param {Object} detail
+ * @return {!Event}
+ */
+export function customEvent(win, type, detail) {
+  if (win.CustomEvent) {
+    return new win.CustomEvent(`slidescroll.${name}`, {detail});
```
Comment (jridgewell) [inline src/utils/event.js]: `slidescroll`?

File: src/utils/event.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  } else {
+    // Deprecated fallback for IE.
+    const e = win.document.createEvent('CustomEvent');
+    e.initCustomEvent(
+        type, /* canBubble */ true, /* cancelable */ true, detail);
```
Comment (jridgewell) [inline src/utils/event.js]: MDN [lists](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#Polyfill) both as being false by default.

File: extensions/amp-selector/0.1/amp-selector.js [hunk 3]
```diff
@@ -208,11 +209,10 @@ export class AmpSelector extends AMP.BaseElement {
         // 'targetOption' - option value of the selected or deselected element.
         // 'selectedOptions' - array of option values of selected elements.
         const name = 'select';
-        const detail = {
+        const selectEvent = customEvent(this.win, `amp-selector.${name}`, {
```
Comment (kmh287) [inline extensions/amp-selector/0.1/amp-selector.js]: won't `name` always be 'select' here?

File: src/utils/event.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+/**
+ * Returns a CustomEvent with a given type and detail; supports fallback for IE.
+ * @param {!Window} win
+ * @param {string} type
+ * @param {Object} detail
+ * @return {!Event}
+ */
+export function customEvent(win, type, detail) {
```
Comment (kmh287) [inline src/utils/event.js]: `createCustomEvent`?

File: extensions/amp-carousel/0.1/slidescroll.js [hunk 5]
```diff
@@ -567,8 +568,8 @@ export class AmpSlideScroll extends BaseSlides {
     this.showSlide_(newIndex);
 
     const name = 'slideChange';
-    const detail = {index: newIndex};
-    const event = new CustomEvent(`slidescroll.${name}`, {detail});
+    const event =
+        customEvent(this.win, `slidescroll.${name}`, {index: newIndex});
```
Comment (kmh287) [inline extensions/amp-carousel/0.1/slidescroll.js]: won't `name` always be slideChange here? If so, any idea why the name was being formatted into the type?

File: extensions/amp-selector/0.1/amp-selector.js (hunk 3 above)
Comment (choumx) [inline extensions/amp-selector/0.1/amp-selector.js]: Yes, but `name` is also used in `trigger()` below.

File: src/utils/event.js (hunk 4 above)
Comment (choumx) [inline src/utils/event.js]: Done.

File: src/utils/event.js (hunk 1 above)
Comment (choumx) [inline src/utils/event.js]: Woops, done.

File: src/utils/event.js (hunk 2 above)
Comment (choumx) [inline src/utils/event.js]: Done.

File: src/utils/event.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+/**
+ * Returns a CustomEvent with a given type and detail; supports fallback for IE.
+ * @param {!Window} win
+ * @param {string} type
+ * @param {Object} detail
+ * @return {!Event}
+ */
+export function createCustomEvent(win, type, detail) {
```
Comment (aghassemi) [inline src/utils/event.js]: can go in `event-helper.js` instead of a new file

Comment (choumx) [inline src/utils/event.js]: Fair point. Done.

**Pull request merged**